glide-core/redis-rs/redis/src/aio/multiplexed_connection.rs

```diff
@@ -175,15 +178,30 @@ where
 
         match &mut entry.response_aggregate {
             ResponseAggregate::SingleCommand => {
-                entry.output.send(result).ok();
+                entry
+                    .output
+                    .send(result.and_then(|v| v.extract_error()))
+                    .ok();
             }
             ResponseAggregate::Pipeline {
                 expected_response_count,
                 current_response_count,
                 buffer,
                 first_err,
+                is_transaction,
             } => {
                 match result {
+                    Ok(Value::ServerError(err))
+                        if *current_response_count < (*expected_response_count - 1)
+                            && *is_transaction =>
+                    {
+                        // In transactions, `count` is always 1 because the final result is a single array (`offset + count = expected_response_count`).
+                        // If `current_response_count < (expected_response_count - 1)`, it means an error occurred before `EXEC` was sent.
```

Comment:
If we received an error and `current_response_count < (expected_response_count - 1)`,...